server/py/services/api/crud/alerts.py

```diff
@@ -215,6 +215,9 @@ def process_event(
                 AlertNotificationPusher().push(alert, event_data)
 
                 if alert.reset_policy == "auto":
+                    services.api.crud.AlertActivation().store_alert_activation(
+                        session, alert, event_data
+                    )
                     self.reset_alert(session, alert.project, alert.name)
```

Comment:
The idea was to write a db record on alert deactivation, so I think even here it's important to follow this order 
```suggestion
                    self.reset_alert(session, alert.project, alert.name)
                    services.api.crud.AlertActivation().store_alert_activation(
                        session, alert, event_data
                    )
```